lib/logstash/devutils/rspec/spec_helper.rb

```diff
@@ -47,16 +46,12 @@
   config.include LogStashHelper
   config.extend LogStashHelper
 
-  exclude_tags = { :redis => true, :socket => true, :performance => true, :couchdb => true, :elasticsearch => true, :elasticsearch_secure => true, :export_cypher => true, :integration => true }
+  config.filter_run_excluding LogStashHelper.excluded_tags
 
-  if LogStash::Environment.windows?
-    exclude_tags[:unix] = true
-  else
-    exclude_tags[:windows] = true
+  config.before(:each) do |example|
+    @__current_example_metadata = example.metadata
```

Comment:
I think the rspec `around` hook may be a slightly better match here, since its scope wraps all of the before/after hooks and will ensure we've got it set for the entirety of the example including any/all hooks.

~~~ ruby
  config.around(:each) do |example|
    @__current_example_metadata = example.metadata
    example.run
  end
~~~